### PR TITLE
Feature/workallocation eui 2894 cancel

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/test/pact/pact-tests/work-allocation-post-cancel.spec.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  })
+
+  // Write Pact when all tests done
+  after(() => provider.finalize())
+
+  describe('when requested to cancel a task', () => {
+    before(() =>
+      provider.addInteraction({
+        state: 'task is cancelled',
+        uponReceiving: 'a request for completion',
```
Comment (paul-graham) [inline api/test/pact/pact-tests/work-allocation-post-cancel.spec.ts:55]: "completion" should be "cancellation"

File: api/test/pact/pact-tests/work-allocation-post-cancel.spec.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+      const { status } = await handleTaskPost(taskUrl, payload, {} as EnhancedRequest)
+      expect(status).equal(200)
+    })
+  })
+
+  describe('when requested to cancel a task that is already cancel', () => {
+    before(() =>
+      provider.addInteraction({
+        state: 'task was already cancel',
+        uponReceiving: 'a request for completion when already cancel',
```
Comment (paul-graham) [inline api/test/pact/pact-tests/work-allocation-post-cancel.spec.ts:80]: "completion" again.

**Review by paul-graham — CHANGES_REQUESTED** [review]
There are a few places where it says "completion" when it should say "cancellation". I don't know if this has an impact on the generation of the PACT interactions but they should be changed for the sake of clarity anyway.

File: api/test/pact/pact-tests/work-allocation-post-cancel.spec.ts (hunk 1 above)
Comment (udaydenduluri33) [reply api/test/pact/pact-tests/work-allocation-post-cancel.spec.ts]: Done

File: api/test/pact/pact-tests/work-allocation-post-cancel.spec.ts (hunk 2 above)
Comment (udaydenduluri33) [reply api/test/pact/pact-tests/work-allocation-post-cancel.spec.ts]: Done

**Review by paul-graham — APPROVED** [review]

**Pull request merged**